pkg/release/condition_set.go

```diff
@@ -124,16 +124,15 @@ func (c *conditionSet) PodNotFound(ctx context.Context, namespace, labelSelector
 	return func() error {
 		o := func() error {
 			pods, err := c.k8sClient.CoreV1().Pods(namespace).List(metav1.ListOptions{LabelSelector: labelSelector})
-			if apierrors.IsNotFound(err) {
```

Comment:
I missed that this is `List` and not `Get` so this error handling isn't appropriate.